src/s_print.c

```diff
@@ -48,14 +48,6 @@ char* pdgui_strnescape(char *dst, size_t dstlen, const char *src, size_t srclen)
     return dst;
 }
 
-static char* strnpointerid(char *dest, const void *pointer, size_t len)
-{
-    *dest=0;
-    if (pointer)
-        snprintf(dest, len, ".x%lx", (unsigned long)pointer);
-    return dest;
-}
-
 static void dopost(const char *s)
```

Comment:
i'm not super-excited by this change :frowning_face: 

instead i think it would be better to use a canvas-to-guiID wrapper everywhere, so there's a single documented way to calculate the ID (rather than having assuming it in multiple places) - even if it is a bit slower.

Reply:
Well, it's a private function and only used in two places. It has been introduced quietly without discussion.

In the future we might refactor the whole codebase and replace all those `.x%lx` strings with something else. But even then it isn't clear to me that a function would be the best approach. This could be also solved with a macro:

```
#define CANVAS_ID ".x%lx.c"
#define OBJECT_ID ".x%lx"

sys_vgui(CANVAS_ID " delete GOP\n",  glist_getcanvas(x));
```

That's why I deleted the function for now. We can later reintroduce it *if* we agree it's really the best approach.

Reply:
Ha, true! The culprit was HCS :-)

I've now pushed the commit to develop and added an issue about fixing the pointer ID problem: https://github.com/pure-data/pure-data/issues/1474